libs/heattracesidesheet/src/lib/ui-sidesheet/HeattraceSidesheet.tsx

```diff
@@ -34,10 +50,35 @@ type HeatTraceProps = {
 };
 
 const HeattraceSidesheetComponent = (props: Required<HeatTraceProps>) => {
-  const circuitDiagramTab = useCircuitDiagramTab(props.item);
-  const workorderTab = useWorkorderTab(props.id);
-  const checklistTab = useChecklistTab(props.id);
-  const threeDTab = use3DTab(props.id);
+  const [activeTab, setActiveTab] = useState(0);
+  const ref = useRef<HTMLDivElement | null>(null);
+  const handleChange = (index: number) => {
+    setActiveTab(index);
+    ref && ref.current && ref.current.scrollTo({ left: index ** index });
+  };
+
+  const { width: sidesheetWidth, setWidth: setSidesheetWidth } = useResizeContext();
+  const reszied = useRef({ hasResized: false, id: props.id });
```

Comment:
We could consider deconstructing the props, so that we don't  needs to do props.xxx everywhere.